cmd/add.go

```diff
@@ -294,6 +315,82 @@ func fileDirIsInDirs(file string, dirSet map[string]bool) bool {
 	return false
 }
 
+// handleMeta takes the user provided meta and the backup meta inputs and
+// returns a map containing all valid metadata.
+func handleMeta(meta, reason, review, removal string) map[string]string {
+	userMeta := parseMetaString(meta)
+
+	backupMeta, err := createBackupMetadata(reason, review, removal)
```

Comment:
I would recommend passing in the `userMeta` map to `createBackupMetadata` and have it set the data directly on it; wouldn't need the copying code below that way.